src/FileCommandFacility.cpp

```diff
@@ -0,0 +1,262 @@
+/** @file FileCommands.cpp
+
+    Fixme: the local "streams" types supporting FileCommandSource are
+    general and could be useful broken out.
+
```

Comment:
Why this is file does not have a dedicated header? 
It even contains a virtual class. Otherwise I don't see the reason of the hierarchy.